sdk/servicebus/service-bus/src/diagnostics/messageSpan.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ConnectionConfig } from "@azure/core-amqp";
+import { getTracer } from "@azure/core-tracing";
+import { Span, SpanContext, SpanKind } from "@opentelemetry/api";
+
+/**
+ * @internal
+ * @ignore
+ */
+export function createMessageSpan(
```

Comment:
Feedback not for this PR, but a note for both Event Hubs & Service Bus for which we can log a generic issue on improvements and make changes for both packages at once in the future:

- The name `createMessageSpan()` is slightly misleading as it may indicate that I can use it at any time when a message is involved. In reality, this method is tied to the "producer" kind, so is usable only when sending message.
- Also, jsdoc for this method would be helpful
- Why is config optional here? I would imagine that we would always have a ConnectionConfig

Reply:
Yes to the first two!

> Why is config optional here? I would imagine that we would always have a ConnectionConfig

parentSpan is not a required parameter but feels primary, I believe that was the reason Chris made the config optional in event-hubs. I can make it the first param in both service-bus and event-hubs and make it required in a later PR.

Reply:
Logged https://github.com/Azure/azure-sdk-for-js/issues/11687